jetty-http/src/test/java/org/eclipse/jetty/http/HttpParserTest.java

```diff
@@ -2295,7 +2310,13 @@ public int getHeaderCacheSize()
         }
 
         @Override
-        public void onComplianceViolation(HttpCompliance compliance, HttpComplianceSection violation, String reason)
+        public boolean isHeaderCacheCaseSensitive()
+        {
+            return _headerCacheCaseSensitive;
+        }
+
+        @Override
+        public void onComplianceViolation(HttpCompliance compliance, HttpCompliance.Violation violation, String reason)
```

Comment:
This isn't limited to HttpCompliance. CookieCompliance can, and should, produce violations as well.
There's also violations produced and reported during Parameter decoding (query / form-data / multipart)

Reply:
OK, so much like your PR we should have `HttpCompliance.Violation` implement a common `Violation` interface which is used for reporting.      Perhaps the `Violation` interface could use a `SpecReference` interface?... or maybe we do that later when we have a usage of SpecReference separate from a Violation.

Reply:
done